src/NHibernate/Linq/Functions/HasFlagGenerator.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq.Expressions;
+using System.Reflection;
+using NHibernate.Hql.Ast;
+using NHibernate.Linq.Visitors;
+
+namespace NHibernate.Linq.Functions
+{
+	internal class HasFlagGenerator : BaseHqlGeneratorForMethod, IRuntimeMethodHqlGenerator
```

Comment:
This does not have to be a `IRuntimeMethodHqlGenerator`

This should work:
```csharp
SupportedMethods = new[] { ReflectHelper.FastGetMethod(Enum.HasFlag, default(Enum)) }
```

Reply:
Ok, I tried, it did not work.

Reply:
It's not static. But should work with 
```C#
SupportedMethods = new[] { ReflectHelper.GetMethodDefinition<Enum>(x => x.HasFlag(default)) };
```

Reply:
i see, we used it with IRuntimeMethodHqlGenerator, there was some issue in past but with newer version there is no problem i updated my PR